src/components/TextField/TextField.tsx

```diff
@@ -38,6 +38,16 @@ type Type =
 
 type Alignment = 'left' | 'center' | 'right';
 
+type InputMode =
+  | 'none'
+  | 'text'
+  | 'decimal'
+  | 'numeric'
+  | 'tel'
+  | 'search'
+  | 'email'
+  | 'url';
```

Comment:
TypeScript `HTMLInputElement.inputMode` types it as `string | undefined` which is not great, but at least we wouldn’t need to update that list if more input modes are added.